glue-lwip/arch/cc.h

```diff
@@ -58,7 +58,11 @@ void sntp_set_system_time (uint32_t t);
 
 #include "mem.h" // useful for os_malloc used in esp-arduino's mDNS
 
-#include "glue.h" // include assembly locking macro used below
```

Comment:
I added the defines in this file to avoid duplication because they are needed on both sides.
What was the issue when using `glue.h` ?

Reply:
Okay I traced down the issue. It's that `ARDUINO` gets defined in `glue.h`. This causes an issue for my project that has another cross-platform library that also uses `ARDUINO` as a build select.

Reply:
Maybe we can change like this, what do you think ?

```#ifndef ARDUINO
#define ARDUINO 0
#endif

#ifndef OPENSDK
#define OPENSDK 0
#endif

#if !ARDUINO && !OPENSDK
#error Must defined ARDUINO or OPENSDK
#endif
```

to 

```
#if (!defined(ARDUINO) || !ARDUINO) && (!defined(OPENSDK) || !OPENSDK)
#error Must defined ARDUINO or OPENSDK
#endif
```

Reply:
If you don't mind changing `ARDUINO` to `BLD_LWIP_ARDUINO` and `OPENSDK` to `BLD_LWIP_OPENSDK`, that would resolve the issue for my project. I feel a bit silly requesting to change it just for my project, but the other library is huge and used by several developers. If you're OK with that, I can make the changes and re-do the PR, and would make it easier for me to submit PRs in the future